### DISCUSSION TITLE
How to migrate away from the registry?

### DISCUSSION THREAD

**Discussion by Rukenshia** [cloudfoundry/bosh#2379] [Q&A]
Hi,

for an update of one of our deployments (concourse), we're also upgrading the bosh director. The director itself is created just fine, but I cannot update our deployment because with the new director, deregistration from the registry of the already existing instances fails:

```
Task 18950 | 13:39:48 | Updating instance web: web/bc19f2b4-9530-4807-b95a-ed397439d1a9 (0) (canary) (00:00:58)
                      L Error: CPI error 'Bosh::Clouds::CloudError' with message 'An attempt to delete registry settings has failed for instance_id=i-02d4e2182cb6a5bc6. The registry is disabled.' in 'delete_vm' CPI method (CPI request ID: 'cpi-176537')
```

I'm not sure what to do now - the instances needed the registry because they used the v1 CPI.

**Comment by nouseforaname** [comment]
Have you also updated your CPI to a newer version or just the director?
What are the versions of:

stemcell
director
cpi
